src/rpc/blockchain.cpp

```diff
@@ -1138,7 +1144,26 @@ UniValue getblockchaininfo(const UniValue &params, bool fHelp)
         "verificationprogress", Checkpoints::GuessVerificationProgress(Params().Checkpoints(), chainActive.Tip()));
     obj.pushKV("initialblockdownload", IsInitialBlockDownload());
     obj.pushKV("chainwork", chainActive.Tip()->nChainWork.GetHex());
+    obj.pushKV("size_on_disk", CalculateCurrentUsage());
     obj.pushKV("pruned", fPruneMode);
+    if (fPruneMode)
+    {
+        CBlockIndex *block = chainActive.Tip();
+        assert(block);
+        {
+            READLOCK(cs_mapBlockIndex);
+            assert(block);
+            while (block && block->pprev && (block->pprev->nStatus & BLOCK_HAVE_DATA))
```

Comment:
why the second assert(block)?

Reply:
a leftover from the port, will remove. Nice catch.

Reply:
done and rebased.